m4/libuv-check-flags.m4

```diff
@@ -73,6 +73,8 @@ AC_DEFUN([CC_CHECK_CFLAG_APPEND], [
 
   AS_IF([eval test x$]AS_TR_SH([cc_cv_cflags_$1])[ = xyes],
     [AM_CFLAGS="$AM_CFLAGS $1"; DEBUG_CFLAGS="$DEBUG_CFLAGS $1"; $2], [$3])
+
+  AC_SUBST([AM_CFLAGS])
```

Comment:
For my understanding: what does the AC_SUBST do that the AM_CFLAGS="..." two lines up doesn't? I believe you when you say it fixes the build error, I'm just trying to understand what it does differently.

Reply:
From my understanding the assign two lines above will just effect the shell that the configure script is running in and not propagate outside of it unless AC_SUBST or similar is being run.

A snippet from the docs:

```
"Create an output variable from a shell variable. Make AC_OUTPUT substitute
the variable variable into output files (typically one or more makefiles)."
```

AC_SUBST also defines a variable for it in addition to the @SUBSTITION@ in the files, so no @@ changes was required in the Makefiles.

So, for me without the AC_SUBST I do not get the AM_CFLAGS assignment in the resulting Makefile:

```
AM_CFLAGS =  -fno-strict-aliasing -g -std=gnu89 -Wall -Wextra -Wno-long-long -Wno-unused-parameter -Wstrict-prototypes
```

Reply:
Okay, thanks for the explanation. CI seems happy so I'll go ahead and merge it.